### ISSUE TITLE
Support for a non-blinking cursor

### ISSUE THREAD

**Issue by cbm755** [jupyter/notebook#1570]
Jupyter has a blinking cursor. Ideally, this should be based on my OS preferences.

At least on my Fedora system, Firefox seems learn this from Gnome.

Does the browser expose this?  Can Jupyter get this property and use it?

(Apologies if this is a dupe, didn't see it...)

**Comment by takluyver** [comment]
I doubt this is something browsers expose. Our text editor component is [CodeMirror](http://codemirror.net/). If that exposes a way to control cursor-blinkitude, we could add a separate setting for it.

**Comment by Carreau** [comment]
yes, codemirror as a [`cursorblinkyness-like`](http://codemirror.net/doc/manual.html#option_cursorBlinkRate) option. 

You can set hat with though your `custom.js` likely.

**Comment by mark-summerfield** [comment]
I certainly can't work with a blinking cursor so hope no blinking becomes a configuration option.

**Comment by tomwhi** [comment]
I really like jupyter notebook, big thanks to the developers. However, I too am afflicted by the cursor blink phenotype. Fortunately someone has explained a solution on their blog:
http://pirsquared.org/blog/notebook-blink.html

At least the last part apparently worked for me - i.e. pasting and running the following javascript code in one of the current notebook's cells:
```
%%javascript
var rate = 0;
// apply setting to  all current CodeMirror instances
IPython.notebook.get_cells().map(
    function(c) {  return c.code_mirror.options.cursorBlinkRate=rate;  }
);

// make sure new CodeMirror instance also use this setting
CodeMirror.defaults.cursorBlinkRate=rate;
```

**Comment by SergeyDidenko** [comment]
You can add `"CodeCell": { "cm_config": {"cursorBlinkRate": 0}}` inside `.jupyter/nbconfig/notebook.json` and restart jupyter notebook for cursor to stop blinking:

A piece of my config:

```javascript
{
  ...
  "CodeCell": {
    "cm_config": {
      "autoCloseBrackets": false,
      "cursorBlinkRate": 0
    }
  },
  ...
}
```

**Comment by al-yakubovich** [comment]
@SergeyDidenko 
Seem to be a great solution but I cannot find .jupyter/nbconfig/notebook.json for Anaconda distribution.

**Comment by hugadams** [comment]
Just another use case for this - I work remote using anydesk and the inherent latency in a  remote session makes the blinking cursor quite wonky and hard to follow

**Comment by gloryVine** [comment]
This is also a huge problem for me. The blinking cursor is incredibly distracting, especially when using the dark theme.

**Comment by appomsk** [comment]
I was able to remove the blinking cursor from jupyter labs for google chrome using Stylus extension by adding for `http://localhost:8888/lab`:

``` css
.CodeMirror-cursor {
  border-left: 2px solid black;
  border-right: none;
  width: 0;
  visibility: visible;
}
```

There is the same extension for Firefox, so I think it should work for this browser too.

>  _A little grunt_: The blinking cursor is not good for any people - just google "Lavender Town Syndrome", but for some, for me, for example, it makes it impossible to use the software. This also leads to some thoughts about the developers of this product, especially when there is a possibility of setting (codemirror allows to set this), but this possibility has not been used for some reason.

**Comment by trondsg** [comment]
Userstyle to remove blink in JupyterLab:
```
@-moz-document url-prefix("http://localhost:8888/lab") {
    @keyframes cm-blink {50% {opacity: 1 !important;}}
    @keyframes cm-blink2 {50% {opacity: 1 !important;}}
}
```

**Comment by AlexFeng863** [comment]
in the new version of jupter lab, you can click Settings -> Settings Editor -> CodeMirror -> Cursor blinking rate, set Cursor blinking rate as 0

**Comment by Carreau** [comment]
> in the new version of jupter lab, you can click Settings -> Settings Editor -> CodeMirror -> Cursor blinking rate, set Cursor blinking rate as 0

Great, closing then.
